// pcln-design-system/dist/Absolute.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(['\n  position: absolute;\n  ', ' ', ' ', ' ', '\n  ', '\n'], ['\n  position: absolute;\n  ', ' ', ' ', ' ', '\n  ', '\n']);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Box = require('./Box');

var _Box2 = _interopRequireDefault(_Box);

var _styledSystem = require('styled-system');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var top = (0, _styledSystem.responsiveStyle)({
  prop: 'top',
  cssProperty: 'top',
  numberToPx: true
});

var bottom = (0, _styledSystem.responsiveStyle)({
  prop: 'bottom',
  cssProperty: 'bottom',
  numberToPx: true
});

var left = (0, _styledSystem.responsiveStyle)({
  prop: 'left',
  cssProperty: 'left',
  numberToPx: true
});

var right = (0, _styledSystem.responsiveStyle)({
  prop: 'right',
  cssProperty: 'right',
  numberToPx: true
});

var zIndex = (0, _styledSystem.style)({
  prop: 'zIndex',
  cssProperty: 'z-index',
  numberToPx: false
});

var Absolute = (0, _styledComponents2.default)(_Box2.default)(_templateObject, top, bottom, left, right, zIndex);

var numberStringOrArray = _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string, _propTypes2.default.array]);

Absolute.propTypes = {
  top: numberStringOrArray,
  bottom: numberStringOrArray,
  left: numberStringOrArray,
  right: numberStringOrArray,
  zIndex: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string])
};

Absolute.displayName = 'Absolute';

exports.default = Absolute;
