// pcln-design-system/dist/Banner.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Box = require('./Box');

var _Box2 = _interopRequireDefault(_Box);

var _Flex = require('./Flex');

var _Flex2 = _interopRequireDefault(_Flex);

var _Text = require('./Text');

var _Text2 = _interopRequireDefault(_Text);

var _Icon = require('./Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _CloseButton = require('./CloseButton');

var _CloseButton2 = _interopRequireDefault(_CloseButton);

var _Heading = require('./Heading');

var _Heading2 = _interopRequireDefault(_Heading);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var bannerColors = {
  green: {
    backgroundColor: 'green',
    color: 'white',
    icon: 'success'
  },
  lightGreen: {
    backgroundColor: 'lightGreen',
    color: 'darkGreen',
    icon: 'success'
  },
  red: {
    backgroundColor: 'red',
    color: 'white',
    icon: 'warning'
  },
  lightRed: {
    backgroundColor: 'lightRed',
    color: 'darkRed',
    icon: 'warning'
  },
  orange: {
    backgroundColor: 'orange',
    color: 'white',
    icon: 'attention'
  },
  lightOrange: {
    backgroundColor: 'lightOrange',
    color: 'darkOrange',
    icon: 'attention'
  },
  blue: {
    backgroundColor: 'blue',
    color: 'white',
    icon: 'information'
  },
  lightBlue: {
    backgroundColor: 'lightBlue',
    color: 'darkBlue',
    icon: 'information'
  }
};

var Banner = function Banner(props) {
  var bannerColor = bannerColors[props.bg] || {};
  var icon = props.iconName || bannerColor.icon;

  return _react2.default.createElement(
    _Box2.default,
    _extends({}, props, {
      bg: bannerColor.backgroundColor || props.bg,
      color: bannerColor.color || props.color
    }),
    _react2.default.createElement(
      _Flex2.default,
      { justify: 'space-between', align: 'flex-start' },
      !!icon && !!props.showIcon && _react2.default.createElement(_Icon2.default, { name: icon, mr: 2, size: 24, mt: '-2px' }),
      _react2.default.createElement(
        _Box2.default,
        { w: 1 },
        _react2.default.createElement(
          _Text2.default,
          { align: props.textAlign },
          _react2.default.createElement(
            _Heading2.default.h5,
            null,
            props.header
          ),
          _react2.default.createElement(
            _Text2.default.span,
            { fontSize: 1 },
            props.text
          ),
          props.children
        )
      ),
      !!props.onClose && _react2.default.createElement(_CloseButton2.default, {
        onClick: props.onClose,
        ml: 2,
        size: 24,
        title: 'close',
        mt: '-2px'
      })
    )
  );
};

Banner.displayName = 'Banner';

Banner.propTypes = {
  header: _propTypes2.default.string,
  iconName: _propTypes2.default.string,
  onClose: _propTypes2.default.func,
  showIcon: _propTypes2.default.bool,
  text: _propTypes2.default.string,
  textAlign: _propTypes2.default.string
};

Banner.defaultProps = {
  bg: 'green',
  textAlign: 'left',
  showIcon: true
};

exports.default = Banner;
