// pcln-design-system/dist/Checkbox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _templateObject = _taggedTemplateLiteral(['\n  display: inline-block;\n  position: relative;\n  vertical-align: middle;\n  cursor: pointer;\n  color: ', ';\n\n  svg[data-name=\'checked\'] {\n    display: none;\n  }\n\n  > input:checked {\n    & ~ svg[data-name=\'checked\'] {\n      display: inline-block;\n      color: ', ';\n    }\n\n    & ~ svg[data-name=\'empty\'] {\n      display: none;\n    }\n  }\n'], ['\n  display: inline-block;\n  position: relative;\n  vertical-align: middle;\n  cursor: pointer;\n  color: ', ';\n\n  svg[data-name=\'checked\'] {\n    display: none;\n  }\n\n  > input:checked {\n    & ~ svg[data-name=\'checked\'] {\n      display: inline-block;\n      color: ', ';\n    }\n\n    & ~ svg[data-name=\'empty\'] {\n      display: none;\n    }\n  }\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  appearance: none;\n  opacity: 0;\n  position: absolute;\n  z-index: 0;\n'], ['\n  appearance: none;\n  opacity: 0;\n  position: absolute;\n  z-index: 0;\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _theme = require('./theme');

var _theme2 = _interopRequireDefault(_theme);

var _ = require('..');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Checkbox = function Checkbox(props) {
  var disabled = props.disabled,
      size = props.size;

  return _react2.default.createElement(
    CheckBoxWrapper,
    { disabled: disabled },
    _react2.default.createElement(StyledInput, _extends({ type: 'checkbox' }, props)),
    _react2.default.createElement(_.Icon, { name: 'boxChecked', size: size, 'data-name': 'checked' }),
    _react2.default.createElement(_.Icon, { name: 'boxEmpty', size: size, 'data-name': 'empty' })
  );
};

var CheckBoxWrapper = _.Box.extend(_templateObject, function (props) {
  return props.disabled ? props.theme.colors.borderGray : props.theme.colors.gray;
}, function (props) {
  return props.disabled ? props.theme.colors.borderGray : props.theme.colors.blue;
});

var StyledInput = _styledComponents2.default.input(_templateObject2);

Checkbox.displayName = 'Checkbox';

Checkbox.propTypes = {
  name: _propTypes2.default.string,
  id: _propTypes2.default.string.isRequired,
  size: _propTypes2.default.number,
  defaultChecked: _propTypes2.default.bool,
  disabled: _propTypes2.default.bool,
  onChange: _propTypes2.default.func.isRequired,
  theme: _propTypes2.default.object
};

Checkbox.defaultProps = {
  size: 20,
  disabled: false,
  defaultChecked: false,
  theme: _theme2.default
};

exports.default = Checkbox;
