// pcln-design-system/dist/CloseButton.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _IconButton = require('./IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _theme = require('./theme');

var _theme2 = _interopRequireDefault(_theme);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CloseButton = function CloseButton(props) {
  return _react2.default.createElement(_IconButton2.default, _extends({}, props, {
    style: { height: props.size + 'px' },
    legacy: false,
    name: 'close'
  }));
};

CloseButton.defaultProps = {
  size: 24,
  title: 'close',
  theme: _theme2.default
};

CloseButton.propTypes = {
  onClick: _propTypes2.default.func,
  size: _propTypes2.default.number,
  title: _propTypes2.default.string
};

CloseButton.displayName = 'CloseButton';

exports.default = CloseButton;
