// pcln-design-system/dist/Flag.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _templateObject = _taggedTemplateLiteral(['\n  width: 8px;\n  height: 8px;\n  align-self: flex-end;\n  ', ';\n  position: absolute;\n  bottom: 0;\n'], ['\n  width: 8px;\n  height: 8px;\n  align-self: flex-end;\n  ', ';\n  position: absolute;\n  bottom: 0;\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  flex: none;\n  background-color: ', ';\n  border-radius: 0 ', ' ', ' 0;\n  /* for 32 x 8 triangle */\n  transform: skew(-14deg);\n  position: relative;\n  z-index: 1;\n'], ['\n  flex: none;\n  background-color: ', ';\n  border-radius: 0 ', ' ', ' 0;\n  /* for 32 x 8 triangle */\n  transform: skew(-14deg);\n  position: relative;\n  z-index: 1;\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n  font-size: ', 'px;\n  border-radius: 0 0 ', ' 0;\n  ', ' ', ';\n  z-index: 2;\n'], ['\n  font-size: ', 'px;\n  border-radius: 0 0 ', ' 0;\n  ', ' ', ';\n  z-index: 2;\n']),
    _templateObject4 = _taggedTemplateLiteral(['\n  position: relative;\n'], ['\n  position: relative;\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require('styled-system');

var _Flex = require('./Flex');

var _Flex2 = _interopRequireDefault(_Flex);

var _Hide = require('./Hide');

var _Hide2 = _interopRequireDefault(_Hide);

var _Box = require('./Box');

var _Box2 = _interopRequireDefault(_Box);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var capitalize = function capitalize(str) {
  return str.charAt(0).toUpperCase() + str.slice(1);
};

var shadowColor = function shadowColor(props) {
  var darkColor = (0, _styledSystem.theme)('colors.dark' + capitalize(props.color))(props);

  return {
    backgroundImage: !darkColor ? '\n        linear-gradient(45deg, transparent 50%, rgba(0, 0, 0, 0.5) 50%),\n        linear-gradient(45deg, transparent 50%, ' + props.color + ' 50%)\n      ' : 'linear-gradient(45deg, transparent 50%, ' + darkColor + ' 50%)'
  };
};

var FlagShadow = (0, _styledComponents2.default)(_Box2.default)(_templateObject, shadowColor);

var FlagRight = (0, _styledComponents2.default)(_Box2.default)(_templateObject2, function (props) {
  return (0, _styledSystem.theme)('colors.' + props.color, props.color)(props);
}, (0, _styledSystem.theme)('radius'), (0, _styledSystem.theme)('radius'));

var flexAuto = function flexAuto(props) {
  return props.flexAuto ? {
    flex: '1 1 auto'
  } : null;
};

var FlagBody = (0, _styledComponents2.default)(_Box2.default)(_templateObject3, (0, _styledSystem.theme)('fontSizes.0'), (0, _styledSystem.theme)('radius'), flexAuto, _styledSystem.color);

var RelativeHide = (0, _styledComponents2.default)(_Hide2.default)(_templateObject4);

var Flag = function Flag(_ref) {
  var color = _ref.color,
      bg = _ref.bg,
      children = _ref.children,
      width = _ref.width,
      props = _objectWithoutProperties(_ref, ['color', 'bg', 'children', 'width']);

  return _react2.default.createElement(
    _Flex2.default,
    _extends({ width: width }, props, { ml: [0, -2] }),
    _react2.default.createElement(
      RelativeHide,
      { xs: true },
      _react2.default.createElement(FlagShadow, { width: '4px', mr: -2, mb: -2, color: bg })
    ),
    _react2.default.createElement(
      FlagBody,
      { flexAuto: !!width, color: color, bg: bg, py: [1, 2], pl: [1, 3] },
      children
    ),
    _react2.default.createElement(FlagRight, { width: '18px', color: bg, ml: -2 })
  );
};

Flag.propTypes = _extends({}, _styledSystem.propTypes.color);

Flag.defaultProps = {
  color: 'white',
  bg: 'green'
};

Flag.displayName = 'Flag';

exports.default = Flag;
