// pcln-design-system/dist/FormField.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _templateObject = _taggedTemplateLiteral(['\n  & ', ' {\n    pointer-events: none;\n  }\n'], ['\n  & ', ' {\n    pointer-events: none;\n  }\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  from {\n    opacity: 0;\n  }\n\n  to {\n    opacity: 1;\n  }\n'], ['\n  from {\n    opacity: 0;\n  }\n\n  to {\n    opacity: 1;\n  }\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _Box = require('./Box');

var _Box2 = _interopRequireDefault(_Box);

var _Flex = require('./Flex');

var _Flex2 = _interopRequireDefault(_Flex);

var _Select = require('./Select');

var _Select2 = _interopRequireDefault(_Select);

var _Icon = require('./Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _Label = require('./Label');

var _Label2 = _interopRequireDefault(_Label);

var _Input = require('./Input');

var _Input2 = _interopRequireDefault(_Input);

var _theme = require('./theme');

var _theme2 = _interopRequireDefault(_theme);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Root = (0, _styledComponents2.default)(_Box2.default)(_templateObject, _Box2.default);

var fadeIn = (0, _styledComponents.keyframes)(_templateObject2);
var labelStyles = {
  animation: fadeIn + ' 0.3s'
};

var getFieldStyles = function getFieldStyles(showLabel) {
  return showLabel ? {
    paddingTop: '20px',
    paddingBottom: '8px',
    transition: 'padding-top 0.1s, padding-bottom 0.1s'
  } : {
    paddingTop: '14px',
    paddingBottom: '14px',
    transition: 'padding-top 0.1s, padding-bottom 0.1s'
  };
};

var noop = function noop() {};

var formElements = [_Input2.default, _Select2.default];

var isFormElement = function isFormElement(element) {
  return formElements.includes(element);
};

var FormField = function (_React$Component) {
  _inherits(FormField, _React$Component);

  function FormField() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, FormField);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = FormField.__proto__ || Object.getPrototypeOf(FormField)).call.apply(_ref, [this].concat(args))), _this), _this.handleChange = function (onChange) {
      return function (e) {
        _this.props.onChange(e);
        if (typeof onChange !== 'function') return;
        onChange(e);
      };
    }, _this.hasValue = function () {
      var children = _this.props.children;

      return _react2.default.Children.toArray(children).reduce(function (a, child) {
        return a || child && isFormElement(child.type) && child.props.value;
      }, false);
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  // for backwards-compatibility


  _createClass(FormField, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          label = _props.label,
          icon = _props.icon,
          children = _props.children,
          onChange = _props.onChange,
          props = _objectWithoutProperties(_props, ['label', 'icon', 'children', 'onChange']);

      var FieldChild = void 0;
      var position = -1;
      var LabelChild = void 0;
      var BeforeIcon = void 0;
      var AfterIcon = void 0;
      var fieldId = void 0;
      var fieldPlaceholder = void 0;
      var iconAdjustment = void 0;

      _react2.default.Children.forEach(children, function (child, index) {
        if (!child) return;

        var type = child.type,
            props = child.props;

        switch (type) {
          case _Label2.default:
            LabelChild = child;
            break;
          case _Input2.default:
          case _Select2.default:
            position = index;
            FieldChild = child;
            fieldId = props.id;
            // For aria-label when Label child is not rendered
            fieldPlaceholder = props.placeholder;
            break;
          case _Icon2.default:
            if (position < 0) {
              BeforeIcon = child;
              iconAdjustment = props.size - 24;
            } else {
              AfterIcon = child;
            }
            break;
        }
      });

      // Handle old version on component's api
      if (icon) {
        AfterIcon = _react2.default.createElement(_Icon2.default, { name: icon });
      }
      if (label) {
        LabelChild = _react2.default.createElement(
          _Label2.default,
          null,
          label
        );
      }
      if (!FieldChild) {
        FieldChild = _react2.default.createElement(_Input2.default, null);
      }

      var showLabel = LabelChild && LabelChild.props.hidden ? false : this.props.alwaysShowLabel || LabelChild && this.hasValue();

      return _react2.default.createElement(
        Root,
        null,
        showLabel && _react2.default.cloneElement(LabelChild, {
          pl: BeforeIcon ? 40 : 2,
          mt: '6px',
          style: labelStyles,
          htmlFor: fieldId
        }),
        _react2.default.createElement(
          _Flex2.default,
          { align: 'center', width: 1, mt: 0 },
          BeforeIcon && _react2.default.createElement(
            _Box2.default,
            {
              mr: -4,
              ml: 8 - iconAdjustment + 'px',
              mt: showLabel ? '-12px' : '2px'
            },
            BeforeIcon
          ),
          _react2.default.cloneElement(FieldChild, _extends({
            'aria-label': !showLabel && fieldPlaceholder ? fieldPlaceholder : null,
            mt: showLabel && -20,
            pl: BeforeIcon ? 40 : 2,
            pr: AfterIcon && 40,
            style: getFieldStyles(showLabel),
            width: 1,
            innerRef: function innerRef(elem) {
              _this2.fieldRef = elem;
            },
            // for backwards compatibility
            onChange: this.handleChange(FieldChild.props.onChange)
          }, props)),
          AfterIcon && _react2.default.createElement(
            _Box2.default,
            { ml: -4, mt: showLabel ? -12 : 2 },
            AfterIcon
          )
        )
      );
    }
  }]);

  return FormField;
}(_react2.default.Component);

FormField.defaultProps = {
  // for backwards-compatibility
  onChange: noop,
  theme: _theme2.default };


FormField.propTypes = {
  alwaysShowLabel: _propTypes2.default.bool,
  children: function children(props, propName, componentName) {
    var prop = props[propName];
    var count = 0;
    var position = 0;
    var labelCount = 0;
    var firstIconPosition = -1;
    var secondIconPosition = 999;
    var iconCount = 0;
    _react2.default.Children.forEach(prop, function (child, index) {
      if (child === null) return;
      switch (child.type) {
        case _Input2.default:
          position = index;
          count++;
          break;
        case _Icon2.default:
          if (iconCount === 0) {
            firstIconPosition = index;
          } else {
            secondIconPosition = index;
          }
          iconCount++;
          break;
        case _Label2.default:
          labelCount++;
          break;
        default:
          return new Error('\'' + child.type + '\' is not a valid child for \'' + componentName + '\'');
      }
    });

    if (!count) {
      return new Error('No \'Input or Select\' child found for \'' + componentName + '\'. Please update your component to use the compound version of this component and pass an Input or Select component as the child');
    }
    if (labelCount > 1) {
      return new Error('Exactly 0 or 1 \'Label\' children should be supplied to \'' + componentName + '\'');
    }
    if (iconCount > 2) {
      return new Error('Up to 2 \'Icon\' children are supported by \'' + componentName + '\'');
    }
    if (iconCount === 2 && (firstIconPosition > position || secondIconPosition < position)) {
      return new Error('If 2 \'Icons\' are provided, the \'Field\' component must be positioned between them as children of \'' + componentName + '\'');
    }
  }
};

exports.default = FormField;
