// pcln-design-system/dist/legacyTheme.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.colors = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _theme = require('./theme');

var _theme2 = _interopRequireDefault(_theme);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var black = '#000';
var white = '#fff';
var text = '#333';
var blue = '#0a84c1';
var green = '#3c910e';
var orange = '#ffaf00';
var red = '#df000f';
var gray = '#666';

var blue0 = '#cee6f3';
var blue1 = '#9dcee6';
var blue2 = '#6cb5da';
var blue3 = '#3b9dcd';

var gray0 = '#d9d9d9';
var gray1 = '#ccc';
var gray2 = '#999';
var gray3 = '#666';

var green0 = '#ecf9e5';
var green1 = '#b1d39f';
var green2 = '#8abd6e';
var green3 = '#63a73e';

var red0 = '#f9cccf';
var red1 = '#f2999f';
var red2 = '#ec666f';
var red3 = '#e5333f';

var orange0 = '#ffefcc';
var orange1 = '#ffdf99';
var orange2 = '#ffcf66';
var orange3 = '#ffbf33';

var colors = exports.colors = {
  black: black,
  white: white,
  text: text,
  blue: blue,
  gray: gray,
  green: green,
  orange: orange,
  red: red,

  blue0: blue0,
  blue1: blue1,
  blue2: blue2,
  blue3: blue3,

  gray0: gray0,
  gray1: gray1,
  gray2: gray2,
  gray3: gray3,

  green0: green0,
  green1: green1,
  green2: green2,
  green3: green3,

  red0: red0,
  red1: red1,
  red2: red2,
  red3: red3,

  orange0: orange0,
  orange1: orange1,
  orange2: orange2,
  orange3: orange3
};

var legacyTheme = _extends({}, _theme2.default, { colors: colors });
exports.default = legacyTheme;
