// pcln-design-system/dist/RatingBadge.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _templateObject = _taggedTemplateLiteral(['\n  display: inline-block;\n  line-height: 1.5;\n  ', ' ', ';\n'], ['\n  display: inline-block;\n  line-height: 1.5;\n  ', ' ', ';\n']);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require('styled-system');

var _Box = require('./Box');

var _Box2 = _interopRequireDefault(_Box);

var _theme = require('./theme');

var _theme2 = _interopRequireDefault(_theme);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var RatingBadge = (0, _styledComponents2.default)(_Box2.default)(_templateObject, _styledSystem.fontWeight, _styledSystem.borderRadius);

RatingBadge.defaultProps = {
  fontWeight: 'bold',
  px: 2,
  color: 'white',
  bg: 'orange',
  borderRadius: 1,
  theme: _theme2.default
};

RatingBadge.propTypes = _extends({}, _styledSystem.propTypes.fontWeight, _styledSystem.propTypes.borderRadius);

exports.default = RatingBadge;
