// pcln-design-system/dist/Select.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(['\n  pointer-events: none;\n'], ['\n  pointer-events: none;\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  appearance: none;\n  display: block;\n  width: 100%;\n  font-family: inherit;\n  color: inherit;\n  background-color: transparent;\n  border-radius: ', ';\n  border-width: 1px;\n  border-style: solid;\n  border-color: ', ';\n  ', ' ', ' &:focus {\n    outline: none;\n    border-color: ', ';\n    box-shadow: 0 0 0 1px ', ';\n  }\n'], ['\n  appearance: none;\n  display: block;\n  width: 100%;\n  font-family: inherit;\n  color: inherit;\n  background-color: transparent;\n  border-radius: ', ';\n  border-width: 1px;\n  border-style: solid;\n  border-color: ', ';\n  ', ' ', ' &:focus {\n    outline: none;\n    border-color: ', ';\n    box-shadow: 0 0 0 1px ', ';\n  }\n']),
    _templateObject3 = _taggedTemplateLiteral([''], ['']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require('styled-system');

var _styledSystem2 = _interopRequireDefault(_styledSystem);

var _theme = require('./theme');

var _theme2 = _interopRequireDefault(_theme);

var _Flex = require('./Flex');

var _Flex2 = _interopRequireDefault(_Flex);

var _Icon = require('./Icon');

var _Icon2 = _interopRequireDefault(_Icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var ClickableIcon = _Icon2.default.extend(_templateObject);

var SelectBase = _styledComponents2.default.select(_templateObject2, _styledSystem2.default.theme('radius'), _styledSystem2.default.theme('colors.borderGray'), _styledSystem.space, _styledSystem.fontSize, _styledSystem2.default.theme('colors.blue'), _styledSystem2.default.theme('colors.blue'));
SelectBase.defaultProps = {
  theme: _theme2.default,
  fontSize: 1,
  m: 0,
  pl: 12,
  pr: 32,
  py: 14
};

var Select = (0, _styledComponents2.default)(function (props) {
  return _react2.default.createElement(
    _Flex2.default,
    { width: 1, align: 'center' },
    _react2.default.createElement(SelectBase, props),
    _react2.default.createElement(ClickableIcon, { legacy: false, ml: -32, name: 'chevronDown', color: 'gray' })
  );
})(_templateObject3);

exports.default = Select;
