// pcln-design-system/dist/Stamp.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _templateObject = _taggedTemplateLiteral(['\n  display: inline-flex;\n  align-items: center;\n  vertical-align: top;\n  min-height: 24px;\n  font-weight: 600;\n  text-transform: uppercase;\n  letter-spacing: ', ';\n  border-radius: 2px;\n  border-width: 1px;\n  border-style: solid;\n  border-color: ', ';\n  ', ' ', ' ', ';\n'], ['\n  display: inline-flex;\n  align-items: center;\n  vertical-align: top;\n  min-height: 24px;\n  font-weight: 600;\n  text-transform: uppercase;\n  letter-spacing: ', ';\n  border-radius: 2px;\n  border-width: 1px;\n  border-style: solid;\n  border-color: ', ';\n  ', ' ', ' ', ';\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _styledSystem = require('styled-system');

var _theme = require('./theme');

var _theme2 = _interopRequireDefault(_theme);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Stamp = _styledComponents2.default.div(_templateObject, (0, _styledSystem.theme)('letterSpacings.caps'), function (props) {
  return _theme2.default.colors.borderGray;
}, _styledSystem.space, _styledSystem.fontSize, _styledSystem.color);

Stamp.displayName = 'Stamp';

var numberStringOrArray = _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string, _propTypes2.default.array]);

Stamp.propTypes = {
  bg: _propTypes2.default.string,
  px: numberStringOrArray,
  py: numberStringOrArray
};

Stamp.defaultProps = {
  px: 1,
  py: 0,
  theme: _theme2.default,
  color: 'gray',
  bg: 'lightGray',
  fontSize: 0
};

exports.default = Stamp;
