// pcln-design-system/dist/Tooltip.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _templateObject = _taggedTemplateLiteral(['\n  display: inline;\n  box-shadow: ', ';\n  font-size: ', 'px;\n  position: absolute;\n  border-radius: ', 'px;\n  box-sizing: border-box;\n  background: ', ';\n  text-align: center;\n\n  ', ' ', ' &::after {\n    content: \'\';\n    position: absolute;\n    width: 0;\n    height: 0;\n    border-width: 5px;\n    border-style: solid;\n    border-color: transparent transparent ', '\n      ', ';\n\n    ', ' ', ' ', ' ', ';\n  }\n'], ['\n  display: inline;\n  box-shadow: ', ';\n  font-size: ', 'px;\n  position: absolute;\n  border-radius: ', 'px;\n  box-sizing: border-box;\n  background: ', ';\n  text-align: center;\n\n  ', ' ', ' &::after {\n    content: \'\';\n    position: absolute;\n    width: 0;\n    height: 0;\n    border-width: 5px;\n    border-style: solid;\n    border-color: transparent transparent ', '\n      ', ';\n\n    ', ' ', ' ', ' ', ';\n  }\n']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Box = require('./Box');

var _Box2 = _interopRequireDefault(_Box);

var _theme = require('./theme');

var _theme2 = _interopRequireDefault(_theme);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var arrowShadow = function arrowShadow(props) {
  return props.top ? {
    'box-shadow': '-9.66px 9.66px 8px 0 rgba(0,0,0,0.04), -4px 4px 4px 0 rgba(0,0,0,0.08)'
  } : {
    'box-shadow': '-1.41px 1.41px 1px 0 rgba(0,0,0,0.01), -3.66px 3.66px 8px 0 rgba(0,0,0,0.04)'
  };
};

var arrowAlign = function arrowAlign(props) {
  return props.left ? { left: '16px', 'margin-left': props.top ? 0 : '15px' } : props.center ? { left: '50%', 'margin-left': props.top ? '-7px' : '7px' } : { right: '16px', 'margin-right': props.top ? '5px' : '-10px' };
};

var arrowPosition = function arrowPosition(props) {
  return props.top ? {
    'transform-origin': '0 0',
    transform: 'rotate(-45deg)',
    bottom: '-10px'
  } : {
    'transform-origin': '0 0',
    transform: 'rotate(-225deg)',
    top: '0'
  };
};

var arrow = function arrow(props) {
  return props.top ? {
    'transform-origin': '0 0',
    transform: 'rotate(-45deg)'
  } : {
    'transform-origin': '0 0',
    transform: 'rotate(-225deg)'
  };
};

var tooltipPosition = function tooltipPosition(props) {
  return props.top ? { bottom: '-8px' } : { top: 0 };
};

var tooltipAlign = function tooltipAlign(props) {
  return props.right ? { right: 0 } : props.center ? { left: '50%', width: 'auto', transform: 'translateX(-50%)' } : null;
};

var TooltipContent = (0, _styledComponents2.default)(_Box2.default)(_templateObject, function (_ref) {
  var theme = _ref.theme;
  return theme.boxShadows[1];
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.fontSizes[0];
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.radii[1];
}, function (_ref4) {
  var theme = _ref4.theme,
      bg = _ref4.bg;
  return theme.colors[bg];
}, tooltipPosition, tooltipAlign, function (_ref5) {
  var theme = _ref5.theme,
      bg = _ref5.bg;
  return theme.colors[bg];
}, function (_ref6) {
  var theme = _ref6.theme,
      bg = _ref6.bg;
  return theme.colors[bg];
}, arrow, arrowPosition, arrowAlign, arrowShadow);

var propTypes = {
  children: _propTypes2.default.any.isRequired,
  bg: _propTypes2.default.string,
  color: _propTypes2.default.string,
  bottom: _propTypes2.default.bool,
  top: _propTypes2.default.bool,
  center: _propTypes2.default.bool,
  left: _propTypes2.default.bool,
  right: _propTypes2.default.bool,
  zIndex: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string])
};

var defaultProps = {
  position: 'bottom',
  color: 'text',
  bg: 'white',
  align: 'right',
  theme: _theme2.default,
  zIndex: 'auto'
};

var Tooltip = function Tooltip(_ref7) {
  var children = _ref7.children,
      props = _objectWithoutProperties(_ref7, ['children']);

  return _react2.default.createElement(
    'div',
    { style: { position: 'relative', zIndex: props.zIndex } },
    _react2.default.createElement(
      TooltipContent,
      _extends({ p: 2, mb: 3, mt: 2 }, props),
      children
    )
  );
};

Tooltip.propTypes = propTypes;
Tooltip.defaultProps = defaultProps;

exports.default = Tooltip;
